### PR TITLE
JDK-8251200: False positive messages about missing comments for serialization

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Using the `@serial` tag, a class can explicitly opt-in to being documented on the serialized form page.  However, a class that does not want to opt-in should not be required to have a doc comment, and should not trigger the "missing comment" warning if there is no comment.

The fix is to check for the presence of the comment on a class before analyzing it to see if it contains `@serial` tags.

The code in `Utils.getDocCommentTree0` is slightly refactored to allow a new predicate `hasDocCommentTree` to be added.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8251200](https://bugs.openjdk.java.net/browse/JDK-8251200): False positive messages about missing comments for serialization

### Reviewers
 * [Hannes Wallnöfer](https://openjdk.java.net/census#hannesw) (@hns - **Reviewer**) ⚠️ Review applies to 76b61ce010905d640c84ed6451095df7e8fc64b6

### Download
`$ git fetch https://git.openjdk.java.net/jdk16 pull/70/head:pull/70`
`$ git checkout pull/70`

### CODE REVIEW

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java [hunk 1]
```diff
@@ -2685,35 +2685,24 @@ public TreePath getTreePath(Element e) {
      */
     private final Map<Element, DocCommentInfo> dcTreeCache = new LinkedHashMap<>();
 
+    /**
+     * Checks whether an element has an associated doc copmment.
```
Comment (hns) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java]: typo: copmment

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java [hunk 2]
```diff
@@ -2685,35 +2685,24 @@ public TreePath getTreePath(Element e) {
...
+     * @param element the element
+     * @return {@code true} if the eleme nt has a comment, and false otherwise
```
Comment (hns) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java]: typo: eleme nt

**Review by hns — APPROVED** [review]
Looks good, apart from two typos in a comment.

**Comment by kusrinivasan** [comment]
Looks good, saw the same typos Hannes noted.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java (hunk 1 above)
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java]: Thanks

**Pull request closed without merging**